File: Data Structures & Algorithms/combination-target-sum/submission-0.py
from collections import defaultdict

class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        results = []

        # def combinations(i, cur, total):
        #     if total == target:
        #         results.append(list(cur))
        #         return

        #     if i>= len(nums) or total > target:
        #         return

        #     cur.append(nums[i])
        #     combinations(i, cur, total + nums[i])
        #     cur.pop()
        #     combinations(i + 1, cur, total)

        # combinations(0, [], 0)
        # return results

        def combinations(i, comb, total):
            if total == 0:
                results.append(list(comb))
                return

            if i >= len(nums) or total < 0:
                return

            comb.append(nums[i])
            combinations(i, comb, total - nums[i])
            comb.pop()
            combinations(i + 1, comb, total)

        combinations(0, [], target)
        return results

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int f = 0, b = numbers.Length-1;

        while (f < b) {
            int sum = numbers[f] + numbers[b];

            if (sum > target)
                --b;
            else if (sum < target)
                ++f;
            else
                return new int[] {f+1,b+1};
        }

        return new int[] {0,0};
    }
}
